sphinx-docs/screenshot.py

```diff
@@ -0,0 +1,296 @@
+            args:       a dictionary of arguments passed to the runhandler function
+    """
+    # The alias with its parse function
+    ALIASES = [("LOGIN", _parse_login)]
+
+    if not arg_str:
+        arg_str = ""
+
+    # First check if we've been passed an aliased_action_sequence
+    words = arg_str.split(' ')
+    for e in ALIASES:
+        if words[0] == e[0]:
+            return e[1](*words[1:])
+
+    commands = arg_str.split('|')
+    parsed_commands = reduce(lambda x,y: x+[y] if y else x, map(_parse_command, commands), [])
+    runhandler = "_command_handler"
+    return { "runhandler":  runhandler,
+             "args":        {'commands': parsed_commands}}
+
+def _parse_user_role(arg_str):
+    if arg_str in ["guest", "coach", "admin", "learner"]:
+        return arg_str
+    else:
+        raise NotImplementedError("Unrecognized user-role: %s" % arg_str)
+
+class Screenshot(Image):
+    """ Provides directive to include screenshot based on given options.
+
+    Since it inherits from the Image directive, it can take Image options.
+    """
+    required_arguments = 0
+    optional_arguments = 0
+    has_content = False
+
+    #########################################################
+    # Handlers will be invoked in the run method should     #
+    # both return appropriate nodes and spawn a process     #
+    # to generate the screenshot (once the script is ready).#
+    #########################################################
+    def _login_handler(self, username, password, submit):
+        from_str_arg = { "users": ["guest"], # This will fail if not guest, because of a redirect
+                         "slug": "",
+                         "start_url": "/securesync/login",
+                         "inputs": [{"#id_username":username},
+                                    {"#id_password":password},
+                                   ],
+                         "pages": [],
+                         "notes": "",
+                       }
+        if submit:
+            from_str_arg["inputs"].append({"<submit>":""})
+        from_str_arg["inputs"].append({"<slug>":self.filename})
+        if hasattr(self, "focus_selector"):
+            from_str_arg["focus"] = {}
+            from_str_arg["focus"]["selector"] = self.focus_selector
+            from_str_arg["focus"]["styles"] = FOCUS_CSS_STYLES
+            from_str_arg["notes"] = self.focus_annotation if hasattr(self, "focus_annotation") else ""
+        # Trying to import django.core.management.call_command gets you into some sort of import hell
+        # Apparently due to a circular import, according to Ben Bach.
+        self.env.screenshot_all_screenshots.append({
+            'docname':  self.env.docname,
+            'from_str_arg': from_str_arg,
+        })
+        self.arguments.append(os.path.join("/", "images", self.filename+".png"))
+        os.system("touch %s" % os.path.join("images", self.filename+".png"))
```

Comment:
Migrate to using `subprocess.Popen`, or maybe try using [utime](https://docs.python.org/2/library/os.html#os.utime) (which seems to be the idiomatic way to do `touch` in pure python).